python/ray/data/impl/block_batching.py

```diff
@@ -64,7 +68,13 @@ def batch_block(block: ObjectRef[Block]):
     for block_window in _sliding_window(blocks, prefetch_blocks + 1):
         block_window = list(block_window)
         with stats.iter_wait_s.timer():
-            ray.wait(block_window, num_returns=1, fetch_local=True)
+            context = DatasetContext.get_current()
+            if context.actor_prefetcher_enabled:
```

Comment:
We should only create and use this prefetching actor if we're prefetching blocks.
```suggestion
            if len(block_window) > 1 and context.actor_prefetcher_enabled:
```